src/briefcase/console.py

```diff
@@ -117,8 +165,75 @@ def __call__(self, prompt):
         "Make Console present the same interface as input()"
         if not self.enabled:
             raise InputDisabled()
+        try:
+            return self._input(prompt)
+        except EOFError:
+            raise KeyboardInterrupt
 
-        return self._input(prompt)
+
+class ProgressBar:
```

Comment:
I had considered this as well....and with additional testing, this is actually already a problem when you consider debug logging.

For instance:
```
❯ briefcase run android -d @beePhone -v
>>> 
>>> Running Command:
>>>     /home/user/.briefcase/tools/android_sdk/platform-tools/adb devices -l
>>> 
>>> Running Command:
>>>     /home/user/.briefcase/tools/android_sdk/emulator/emulator -list-avds
>>> 
>>> Running Command:
>>>     /home/user/.briefcase/tools/android_sdk/emulator/emulator -list-avds
Starting emulator beePhone...
>>> 
>>> Running Command:
>>>     /home/user/.briefcase/tools/android_sdk/emulator/emulator @beePhone -dns-server 8.8.8.8

Waiting for emulator to start....>>> 
>>> Running Command:
>>>     /home/user/.briefcase/tools/android_sdk/platform-tools/adb devices -l
.>>> 
>>> Running Command:
>>>     /home/user/.briefcase/tools/android_sdk/platform-tools/adb devices -l
.>>> 
>>> Running Command:
>>>     /home/user/.briefcase/tools/android_sdk/platform-tools/adb devices -l
.>>> 
>>> Running Command:
>>>     /home/user/.briefcase/tools/android_sdk/platform-tools/adb devices -l
>>> 
>>> Running Command:
>>>     /home/user/.briefcase/tools/android_sdk/platform-tools/adb -s emulator-5554 emu avd name

 booting...>>> 
>>> Running Command:
>>>     /home/user/.briefcase/tools/android_sdk/platform-tools/adb -s emulator-5554 shell getprop sys.boot_completed
[TRUNCATED]
```

Reply:
So, any larger solution would also probably need some closer integration between `Log` and `Console`....at least so that `Log` calls could interact with any ongoing progress/wait bars. Perhaps making `Log` and `Console` singletons would make that more straightforward than trying to make them aware of each other's instantiations....absent a much different refactor.

At any rate, exposing the progress and wait bars from `Console` is probably a better starting place for this. I added two new methods in `Console` to simply return an instantiated progress/wait bar.

Reply:
A related problem - creating a progress bar while you're already inside a progress bar. Not likely to be something that is a real problem in practice, but the accounting we'd need to do to track "are we in a progress bar" would also help us identify if there is a need to flush output for debugging, etc.